iavl/iavl_node.go

```diff
@@ -522,5 +552,31 @@ func removeOrphan(t *IAVLTree, node *IAVLNode) {
 	if t.ndb == nil {
 		return
 	}
+
+	// Don't orphan persistent nodes if another tree depends on it
+	if Committed != nil && t != Committed {
```

Comment:
global state? Are you sure this is a good idea? If there is no other way, I guess its fine.